src/distributed_graph_generator.cc

```diff
@@ -450,13 +450,13 @@ void distributed_graph_generator::generate_distributed_commands(const task& tsk)
 			assert(writer_cmd != nullptr);
 
 			// We're only interested in writes that happen within the same task as the push
-			if(isa<task_command>(writer_cmd) && static_cast<task_command*>(writer_cmd)->get_tid() == tsk.get_id()) {
+			if(utils::isa<task_command>(writer_cmd) && utils::as<task_command>(writer_cmd)->get_tid() == tsk.get_id()) {
```

Comment:
This would be cleaner with an if-with-initializer that doesn't need to repeat the target type, i.e.
```c++
if(const auto writer_tcmd = dynamic_cast<const task_command*>(writer_cmd); writer_tcmd && writer_tcmd->get_tid() == tsk.get_id())
```
and even prettier if we had another utility `try_as`:
```c++
template <typename T, typename P>
auto try_as(P* p) {
	return dynamic_cast<std::conditional_t<std::is_const_v<P>, const T*, T*>>(p);
}
```
```c++
if(const auto writer_tcmd = utils::try_as<task_command>(writer_cmd); writer_tcmd && writer_tcmd->get_tid() == tsk.get_id())
```

Reply:
I guess this comes down to preference, to me this doesn't look cleaner at all. Yes, it doesn't repeat the type, but instead introduces a name that is not used within the following block, while also requiring a `nullptr` check (with an implicit conversion too bool, which I'm also not a huge fan of) so the whole thing ends up being longer.

But I agree that a `try_as` function could be useful in cases where we *do* use the name in the following block. I think we have a couple of those as well (currently using `dynamic_cast`).

Reply:
Sure, this does not necessarily need to be addressed in this PR.